types/react-big-calendar/index.d.ts

```diff
@@ -268,7 +268,7 @@ export interface CalendarProps<TEvent extends object = Event, TResource extends
     onDoubleClickEvent?: (event: TEvent, e: React.SyntheticEvent<HTMLElement>) => void;
     onSelectEvent?: (event: TEvent, e: React.SyntheticEvent<HTMLElement>) => void;
     onSelecting?: (range: { start: stringOrDate; end: stringOrDate }) => boolean | undefined | null;
-    onRangeChange?: (range: Date[] | { start: stringOrDate; end: stringOrDate }) => void;
+    onRangeChange?: (range: Date[], view: View) => void;
```

Comment:
This doesn't seem to be what the docs state:

> Returns an Array of dates or an object with start and end dates for BUILTIN views.

-- http://intljusticemission.github.io/react-big-calendar/examples/index.html#prop-onRangeChange

Reply:
https://github.com/intljusticemission/react-big-calendar/blob/master/src/Calendar.js#L253 ◀️ Code comment seems to agree with @catruzz, it says the type is Date only though :/

Reply:
Clicking next/prev repros this on the console: https://codesandbox.io/s/weathered-dawn-4rijx

Reply:
Thanks to the code provided by @PaitoAnderson I found out that `onRangeChange` actually returns an array of Dates or an object having `start` and `end` keys (as stated in the docs) depending on the current view (that is also returned) and Typescript returns errors accordingly.

1. Day: returns array with one element
2. Week: returns array of 7 elements
3. Month: returns object `{start: ..., end: ...}`
4. Agenda: returns object `{start: ..., end: ...}`

Here you can see the working example with the `view` parameter evidence (Typescript is included but doesn't fire any error here though, as it does in my VSCode)
https://codesandbox.io/s/nifty-lake-t99ky